### PR TITLE
Add lsp tests and Toit LSP server.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/lsp/fail.cmake [hunk 1]
```diff
@@ -0,0 +1,96 @@
+set(TOIT_FAILING_TESTS
```
Comment (floitsch) [inline tests/lsp/fail.cmake:1]: In this PR all tests are failing.
Follow-up PR is on its way.

File: tests/fail.cmake [hunk 2]
```diff
@@ -0,0 +1,2 @@
+set(TOIT_FAILING_TESTS
```
Comment (kasperl) [inline tests/fail.cmake:1]: Should we have copyright headers in these files too? tests/CMakeLists.txt has such a header.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/fail.cmake (hunk 2 above)
Comment (floitsch) [reply tests/fail.cmake:1]: done.

**Pull request merged**